### PR TITLE
Create api for filter page

### PR DESCRIPTION
**Pull request by OlegNovatskiy**
In it pull request are review codes for API of component map and create API for component filter

### CODE REVIEW

File: src/main/java/com/declaration/api/components/entity/ComponentFilterDataFilter.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+	private String location;
+	private Integer limit;
+	private Integer page;
+	private String firstName;
+	private String lastName;
+	private String middleName;
+
+	public ComponentFilterDataFilter() {
+		super();
```
Comment (CortesV) [inline src/main/java/com/declaration/api/components/entity/ComponentFilterDataFilter.java]: You need super()?????

Comment (CortesV) [inline src/main/java/com/declaration/api/components/entity/ComponentFilterDataFilter.java]: You need super()???????

File: src/main/java/com/declaration/api/components/entity/ComponentFilterInfo.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package com.declaration.api.components.entity;
+
+import java.io.Serializable;
+
```
Comment (CortesV) [inline src/main/java/com/declaration/api/components/entity/ComponentFilterInfo.java]: java doc

File: src/main/java/com/declaration/api/components/service/ComponentFilterDAO.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+ * 
+ * @author olegnovatskiy
+ */
+@Repository
+public class ComponentFilterDAO implements IComponentFilter {
+
+	@Autowired
+	private JdbcTemplate jdbcTemplate;
+
+	private static final String SELECT_QUERY = "SELECT ty.declaration_id, si.first_name, si.last_name, si.middle_name, si.work_post, si.actual_region FROM type as ty INNER JOIN subject_info AS si ON ty.person_id = si.id";
```
Comment (CortesV) [inline src/main/java/com/declaration/api/components/service/ComponentFilterDAO.java]: SELECT_QUERY incorrect name, I don't understand what do this constant

File: src/main/java/com/declaration/api/components/service/ComponentMapDAO.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ * 
+ * @author olegnovatskiy
+ */
+@Repository
+public class ComponentMapDAO implements IComponentMap {
+
+	@Autowired
+	private JdbcTemplate jdbcTemplate;
+
+	private static final String SELECT_QUERY = "SELECT si.actual_region AS region, COUNT(ty.id) AS count_declaration, SUM(inc.size_income) AS sum_income_in_region, SUM(inc.size_assets) AS sum_assets_in_region FROM type AS ty INNER JOIN subject_info AS si ON si.id = ty.person_id INNER JOIN income AS inc ON inc.person_id = si.id ";
```
Comment (Yagi7337) [inline src/main/java/com/declaration/api/components/service/ComponentMapDAO.java]: SELECT_QUERY incorrect name

File: src/main/java/com/declaration/api/components/service/ComponentMapDAO.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	private static final String GROUP_BY = " GROUP BY si.actual_region;";
```
Comment (Yagi7337) [inline src/main/java/com/declaration/api/components/service/ComponentMapDAO.java]: GROUP_BY incorrect name

File: src/main/java/com/declaration/api/components/entity/ComponentFilterDataFilter.java (hunk 1 above)
Comment (OlegNovatskiy) [reply src/main/java/com/declaration/api/components/entity/ComponentFilterDataFilter.java]: DONE

Comment (OlegNovatskiy) [reply src/main/java/com/declaration/api/components/entity/ComponentFilterDataFilter.java]: DONE

File: src/main/java/com/declaration/api/components/service/ComponentFilterDAO.java (hunk 3 above)
Comment (OlegNovatskiy) [reply src/main/java/com/declaration/api/components/service/ComponentFilterDAO.java]: DONE

File: src/main/java/com/declaration/api/components/entity/ComponentFilterInfo.java (hunk 2 above)
Comment (OlegNovatskiy) [reply src/main/java/com/declaration/api/components/entity/ComponentFilterInfo.java]: DONE

File: src/main/java/com/declaration/api/components/service/ComponentMapDAO.java (hunk 4 above)
Comment (OlegNovatskiy) [reply src/main/java/com/declaration/api/components/service/ComponentMapDAO.java]: DONE

File: src/main/java/com/declaration/api/components/service/ComponentMapDAO.java (hunk 5 above)
Comment (OlegNovatskiy) [reply src/main/java/com/declaration/api/components/service/ComponentMapDAO.java]: DONE

File: src/main/java/com/declaration/api/components/entity/FilterComponentDataFilter.java [hunk 6]
```diff
@@ -8,7 +8,7 @@
  * @author olegnovatskiy
  *
  */
-public class ComponentFilterDataFilter extends ComponentMapDataFilter implements Serializable {
+public class FilterComponentDataFilter extends MapComponentDataFilter implements Serializable {
```
Comment (CortesV) [inline src/main/java/com/declaration/api/components/entity/FilterComponentDataFilter.java]: incorrect name of classes

File: src/main/java/com/declaration/api/components/entity/FilterComponentDataFilter.java [hunk 7]
```diff
@@ -19,16 +19,6 @@
...
-		super();
-		this.location = "";
-		this.limit = 10;
-		this.page = 1;
-		this.firstName = "";
-		this.lastName = "";
-		this.middleName = "";
-	}
-
 	public String getLocation() {
```
Comment (CortesV) [inline src/main/java/com/declaration/api/components/entity/FilterComponentDataFilter.java]: what is mean location?

Comment (CortesV) [reply src/main/java/com/declaration/api/components/entity/FilterComponentDataFilter.java]: bad name for variable location

File: src/main/java/com/declaration/api/components/entity/FilterComponentDeclarations.java [hunk 8]
```diff
@@ -13,7 +18,7 @@
 	private String position;
 	private String location;
```
Comment (CortesV) [inline src/main/java/com/declaration/api/components/entity/FilterComponentDeclarations.java]: bad names because I don't understand what it mean

**Pull request merged**